--- wpmu-global-search/includes/functions.php~	2010-05-19 15:57:49.000000000 +0300
+++ wpmu-global-search/includes/functions.php	2010-05-19 17:11:33.980063897 +0300
@@ -28,8 +28,11 @@
 	return apply_filters( 'get_search_query', stripslashes( get_query_var( 'uws' ) ) );
 }
 
-function wpmu_globalsearch_get_the_content($s) {
+function wpmu_globalsearch_get_the_content($s, $cutoff = 0) {
     $content = $s->post_content;
+    if ($cutoff) {
+        $content = mb_substr($content, 0, $cutoff). '...';
+    }
     apply_filters('the_content', $content);
 
     $output = '';
--- wpmu-global-search/wpmu-global-search.php~	2010-05-19 15:57:49.000000000 +0300
+++ wpmu-global-search/wpmu-global-search.php	2010-05-19 17:13:30.866699686 +0300
@@ -142,6 +142,7 @@
 
     $options = get_option('widget_wp_wpmu_globalsearch');
     $results_limit = empty($options['results_limit']) ? 2 : $options['results_limit'];
+    $cutoff = empty($options['results_cutoff']) ? 500 : $options['results_cutoff'];
 
     if(!empty($wp_query->query_vars['uws'])) {
         $term = $wp_query->query_vars['uws'];
@@ -174,7 +175,7 @@
                     </div>
 
                     <div class="entry">
-                        <?php echo wpmu_globalsearch_get_the_content($s); ?>
+                        <?php echo wpmu_globalsearch_get_the_content($s, $cutoff); ?>
                     </div>
 
                     <div class="postmetadatasearch">
